### PR TITLE
Feature/cr 2968 metadata store refactor

### PR DESCRIPTION
**Pull request by ori-bar**

### CODE REVIEW

**Comment by yi2020** [comment]
@ori-bar why are there rules being removed for all three metadata files?

File: cloudrail/knowledge/rules/rules_metadata_store.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        return self.rules_metadata.get(rule_id)
+
+    def list_rules_metadata(self) -> List[RuleMetadata]:
+        return list(self.rules_metadata.values())
+
+    def add_rules_metadata(self, new_rules_metadata_dict: dict):
+        new_rules_metadata = self._get_rules_metadata_from_dict(new_rules_metadata_dict)
+        self.rules_metadata = {**self.rules_metadata, **new_rules_metadata}
+
+    def merge(self, rule_metadata_store: 'RulesMetadataStore'):
```
Comment (alonashkenazi) [inline cloudrail/knowledge/rules/rules_metadata_store.py:37]: I don't see that someone using it, can you point me where we are using it?

File: cloudrail/knowledge/rules/rules_metadata_store.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
+import os
+from typing import Dict, Optional, List
+
+import yaml
+
+from cloudrail.knowledge.context.cloud_provider import CloudProvider
+from cloudrail.knowledge.rules.rule_metadata import RuleMetadata, RuleSeverity, RuleType, SecurityLayer, ResourceType
+
+RULE_ID = 'rule_id'
```
Comment (alonashkenazi) [inline cloudrail/knowledge/rules/rules_metadata_store.py:9]: Should it be inside the rulemetadata class?

File: cloudrail/knowledge/rules/rules_metadata_store.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            raise Exception('rule id can not be template based')
+
+    @classmethod
+    def _verify_rules(cls, rules):
+        cls._verify_all_fields_filled(rules)
+        cls._verify_name_unique(rules)
+        cls._verify_id_unique(rules)
+        cls._verify_rule_id_not_template(rules)
+
+    def _get_rules_metadata_from_dict(self, raw_data: dict) -> Dict[str, RuleMetadata]:
```
Comment (alonashkenazi) [inline cloudrail/knowledge/rules/rules_metadata_store.py:115]: Maybe this should be an object:Dict[str, RuleMetadata] -> RuleMetaDataStore

File: cloudrail/knowledge/rules/rules_metadata_store.py (hunk 1 above)
Comment (ori-bar) [reply cloudrail/knowledge/rules/rules_metadata_store.py:37]: in the platform (custom rules metadata store)

File: cloudrail/knowledge/rules/rules_metadata_store.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    def __init__(self, rules_metadata_dict: dict):
+        self.rules_metadata = self._get_rules_metadata_from_dict(rules_metadata_dict)
+
+    def get_by_rule_id(self, rule_id: str) -> Optional[RuleMetadata]:
+        return self.rules_metadata.get(rule_id)
+
+    def list_rules_metadata(self) -> List[RuleMetadata]:
+        return list(self.rules_metadata.values())
+
+    def add_rules_metadata(self, new_rules_metadata_dict: dict):
```
[the remark this answers is not in the document]
Comment (ori-bar) [reply cloudrail/knowledge/rules/rules_metadata_store.py:33]: but the whole point is to add metadata to the class and the form of the raw metadata is dict (from yaml)

File: cloudrail/knowledge/rules/rules_metadata_store.py (hunk 2 above)
Comment (ori-bar) [reply cloudrail/knowledge/rules/rules_metadata_store.py:9]: why put constants inside the class?

File: cloudrail/knowledge/rules/rules_metadata_store.py (hunk 3 above)
Comment (ori-bar) [reply cloudrail/knowledge/rules/rules_metadata_store.py:115]: why? it's just internal it's not something we expose externally

File: cloudrail/knowledge/rules/rules_metadata_store.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (alonashkenazi) [reply cloudrail/knowledge/rules/rules_metadata_store.py:33]: shouldn't new_rules_metadata_dict be {key: {RuleMetadata}}?

File: cloudrail/knowledge/rules/rules_metadata_store.py (hunk 1 above)
Comment (alonashkenazi) [reply cloudrail/knowledge/rules/rules_metadata_store.py:37]: So if it's public, shouldn't we have tests?

File: cloudrail/knowledge/rules/rules_metadata_store.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+            cloud_provider=CloudProvider(rule[CLOUD_PROVIDER]),
+            is_deleted=rule.get('is_deleted', False)) for rule in rules}
+
+
+def read_metadata_file(file_path: str):
+    with open(file_path, 'r') as file:
+        return yaml.load(file)
+
+
+class CloudrailRulesMetadataStore(RulesMetadataStore):
```
Comment (alonashkenazi) [inline cloudrail/knowledge/rules/rules_metadata_store.py:138]: I don't think that the class name should have Cloudrail in it. Maye CloudsRulesMetadataStore

**Comment by ori-bar** [comment]
> @ori-bar why are there rules being removed for all three metadata files?

@yi2020 moved the metadata of checkov rules to the platform from the knowledge

**Review by alonashkenazi — APPROVED** [review]

**Pull request merged**